test/pointer_type_mapping.ll

```diff
@@ -0,0 +1,63 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+all -spirv-text -o %t
+; RUN: FileCheck < %t %s
+
+; CHECK: Name [[#NAME:]] "struct._ZTS6Object.Object"
+; CHECK-COUNT-1: TypeStruct [[#NAME]]
+; TODO add check count one and remove unused, when the type mapping bug is fixed
```

Comment:
You could already add the check and XFAIL the test perhaps?

Reply:
Since this tests is also checks that only *single* `TypeStruct` is actually being used, it wouldn't be clear why does it fail: due to one bug or another?

I mean, we have a bigger general problem with that mapping and a particular case where it causes problems. The case is workarounded and we have a test for that. The test is not entirely finished, because the bigger problem is still there.

Reply:
TBH this PR in the current stage is more about highlighting the issue. I would prefer to fix first and push it in the same branch (unless fixing it takes a lot of time).

Reply:
Fair enough, XFAILing this test would mean the work-around is no longer tested indeed.  So fine to leave it as is then.

Reply:
Yet if we want it be merged anyway, I should apply [this](https://github.com/intel/llvm/pull/3856#discussion_r642806327) comment